test/suite/test_checkpoint_snapshot04.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env python
+#
+# Public Domain 2014-present MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import fnmatch, os, shutil, threading, time
+import wiredtiger, wttest
+from wtbackup import backup_base
+from wtdataset import SimpleDataSet
+from wtscenario import make_scenarios
+
+# test_checkpoint_snapshot04.py
+#   Test backup restore when the checkpoint contains data greater than the
+#   checkpoint snapshot.
+class test_checkpoint_snapshot04(backup_base):
+    dir = 'backup.dir'
+
+    # Create a table.
+    uri = "table:test_checkpoint_snapshot04"
+    nrows = 5000
+
+    target_backup = [
+        ('full', dict(target=False)),
+        ('target', dict(target=True))
+    ]
+
+    scenarios = make_scenarios(target_backup)
+
+    def conn_config(self):
+        config = 'cache_size=25MB'
+        return config
+
+    def large_updates(self, uri, value, ds, nrows):
+        # Update a large number of records.
+        session = self.session
+        cursor = session.open_cursor(uri)
+        for i in range(0, nrows):
+            session.begin_transaction()
+            cursor[ds.key(i)] = value
+            session.commit_transaction()
+        cursor.close()
+
+    def check(self, check_value, uri, nrows):
+        session = self.session
+        session.begin_transaction()
+        cursor = session.open_cursor(uri)
+        count = 0
+        for k, v in cursor:
+            self.assertEqual(v, check_value)
+            count += 1
+        session.commit_transaction()
+        self.assertEqual(count, nrows)
+
+    def test_checkpoint_snapshot(self):
```

Comment:
What is this testing? It seems to be populating, taking a checkpoint, taking a backup and then checking that the backup contains the same data as the current database.

I don't really get the relation to write gens. Also, I was under the impression that this change involves rollback to stable, yet there's no restart happening here.

Reply:
As part of comparing the backup it internally using the `wt` utility to restart the database to dump. As part of `wt` utility the recovery occurs.

Reply:
Understood, thanks.